src/myspace/PersonalDetails.jsx

```diff
@@ -18,23 +22,70 @@ export default function PersonalDetails() {
     },
   };
 
-  let data = "";
   useEffect(() => {
     fetch(`${BASE_API}/user/personal_details`, requestPersonalDetails)
       .then(async response => {
-        data = await response.json();
+        const data = await response.json();
         if (response.ok) {
           return setPersonalDetails(data);
-        } else {
-          throw new Error(data.message);
         }
-      })
-      .catch(error => {
         setErrorMessage(data.message);
-      });
+      })
+      .catch(error =>
+        setErrorMessage("The service is temporarily unavailable, please try again later.")
+      )
 
   });
 
+  const handleSubmit = async e => {
+    e.preventDefault();
+
+    let payload = {}
+    new FormData(e.target).forEach((value, key) => {
+      if (key === "need_mentoring" || key === "available_to_mentor")
+          if (value === "true")
```

Comment:
Too many ifs and else's here. Code doesn't look clean with so many conditionals

Reply:
We can refactor this tomorrow together, or give it a shot if you like

Reply:
I For this one, I'll wait till our meeting tonight on your office hour (if you feel well enough). I'm not sure what else to do with this option.